internal/services/resource/resource_provider_registration_resource.go

```diff
@@ -336,6 +336,9 @@ func (r ResourceProviderRegistrationResource) CustomImporter() sdk.ResourceRunFu
 		client := metadata.Client.Resource.ResourceProvidersClient
 		account := metadata.Client.Account
 
+		ctx, cancel := context.WithDeadline(ctx, time.Now().Add(5*time.Minute))
+		defer cancel()
```

Comment:
it'd be more consistent to make this:

```suggestion
		if _, ok := ctx.Deadline(); !ok {
			var cancelFunc context.CancelFunc
			ctx, cancel = context.WithTimeout(ctx, 5*time.Minute)
			defer cancel()
		}
```

As such, could we update this?

Reply:
Thanks, I will change this.